include/circt/Dialect/FIRRTL/FIRRTLTypesImpl.td

```diff
@@ -139,34 +139,36 @@ def AnalogTypeImpl : FIRRTLImplType<"Analog",
   let genVerifyDecl = true;
 }
 
-def FVectorTypeImpl : FIRRTLImplType<"FVector", [FieldIDTypeInterface]> {
+class BaseVectorTypeImpl<string name, string ElementType, list<Trait> traits = [], string BaseType = ElementType> : FIRRTLImplType<name, traits, BaseType> {
   let summary = "a fixed size collection of elements, like an array.";
   let parameters = (ins
-      TypeParameter<"::circt::firrtl::FIRRTLBaseType", "Type of vector elements">:$elementType,
+      TypeParameter<ElementType, "Type of vector elements">:$lementType,
      "size_t":$numElements,
      "bool":$isConst
   );
-  let storageClass = "FVectorTypeStorage";
+  let storageClass = name # "TypeStorage";
   let skipDefaultBuilders = true;
   let builders = [
     TypeBuilderWithInferredContext<(ins
-      "FIRRTLBaseType":$elementType,
+      ElementType:$elementType,
       "size_t":$numElements,
        CArg<"bool", "false">:$isConst)
     >
   ];
   let extraClassDeclaration = [{
-    FIRRTLBaseType getElementType() const;
+    using ElementType = }] # ElementType # [{;
+
+    ElementType getElementType() const;
     size_t getNumElements() const;
 
     /// Return the recursive properties of the type.
     RecursiveTypeProperties getRecursiveTypeProperties() const;
 
     /// Return this type with any flip types recursively removed from itself.
-    FIRRTLBaseType getPassiveType();
+    }] # BaseType # [{ getPassiveType();
```

Comment:
Does `getPassiveType` make sense for open types, considering that ref types are not passive? Is it even needed since open type lowering is the first pass that will be performed?

Reply:
Interesting.  It doesn't seem wrong to ask to remove all `flips` from the open aggregates, but it would result in something where `isPassive()` wouldn't be true, which is strange if not just wrong.

Additionally, I don't think this is reachable as-is, since it's only through FIRRTLBaseType that we invoke that method today.

Will update dropping this and the cached passive type pointer from their storage, thanks!